sequencing-server/src/lib/batchLoaders/parcelBatchLoader.ts

```diff
@@ -0,0 +1,59 @@
+import { ErrorWithStatusCode } from '../../utils/ErrorWithStatusCode.js';
+import type { BatchLoader } from './index.js';
+import { gql, GraphQLClient } from 'graphql-request';
+
+export type Parcel = {
+  id: number;
+  command_dictionary: {
+    id: number;
+  };
+  parameter_dictionary: {
+    parameter_dictionary: {
+      id: number;
+    };
+  }[];
+  channel_dictionary?: {
+    id: number;
+  };
+};
+
+export type Parcels = {
+  parcel: Parcel[];
+};
+
+export const parcelBatchLoader: BatchLoader<{ parcelId: number }, Parcel, { graphqlClient: GraphQLClient }> =
+  opts => async keys => {
+    const { parcel } = await opts.graphqlClient.request<Parcels>(
+      gql`
+        query GetParcels($parcelIds: [Int!]!) {
+          parcel(where: { id: { _in: $parcelIds } }) {
+            id
+            parameter_dictionary(where: { parcel_id: { _in: $parcelIds } }) {
```

Comment:
This `where` is unnecessary: `parameter_dictionary` will fetch the dictionaries associated with the specific parcel.

Reply:
Thank you for the suggestion, I made the change, and my manual test passed.